assets/_common/js/filters.js

```diff
@@ -0,0 +1,345 @@
+            if (targetType === 'agencies') {
+                switch (key) {
+                    case 'organization':
+                        return item.organizations && item.organizations.includes(value);
+                    case 'status':
+                        return true;
+                    case 'language':
+                        return true;
+                    case 'category':
+                        return true;
+                    case 'platform':
+                        return true;
+                    default:
+                        if (Array.isArray(item[key])) {
+                            return item[key].includes(value);
+                        }
+                        return item[key] === value;
+                }   
+            } else {
+                let actualKey = key;
+                
+                switch (key) {
+                    case 'language':
+                        actualKey = 'languages'
+                        break;
+                    case 'category':
+                        actualKey = 'categories'
+                        break;
+                    case 'platform':
+                        actualKey = 'platforms'
+                        break;
+                }
+
+                if (Array.isArray(item[actualKey])) {
+                    return item[actualKey].includes(value);
+                }
+                return item[actualKey] === value;
+            }
+        });
+    }, data);
+};
+
+const sortData = (data, sortBy) => {
+    const [field, direction] = sortBy.split('-');
+
+    return [...data].sort((a, b) => {
+        let aVal, bVal;
+
+        switch (field) {
+            case 'name':
+                aVal = a.name?.toLowerCase() || '';
+                bVal = b.name?.toLowerCase() || '';
+                break;
+            case 'projects':
+                aVal = a.projectCount || 0;
+                bVal = b.projectCount || 0;
+                break;
+            case 'modified':
+                aVal = new Date(a.dates?.lastModified || 0);
+                bVal = new Date(b.dates?.lastModified || 0);
+                break;
+            case 'created':
+                aVal = new Date(a.dates?.created || 0);
+                bVal = new Date(b.dates?.created || 0);
+                break;
+            default:
+                return 0;
+        }
+
+        if (direction === 'desc') {
+            return aVal < bVal ? 1 : aVal > bVal ? -1 : 0;
+        }
+        return aVal > bVal ? 1 : aVal < bVal ? -1 : 0;
+    });
+};
+
+const processFilters = (state) => {
+   let filtered = [...state.originalData];
+   
+   filtered = applySearchFilter(filtered, state.filters.search, state.targetType);
+   filtered = applySelectFilters(filtered, state.filters, state.targetType);
+   filtered = sortData(filtered, state.sortBy);
+
+   return {
+    ...state,
+    filteredData: filtered
+   };
+};
+
+const createAgencyCardHTML = (agency) => `
```

Comment:
If you can fix the errors shown in the html-validation tests, that would be great! https://github.com/DSACMS/code-gov/actions/runs/15407596376/job/43353185736?pr=15